circus/tests/test_client.py

```diff
@@ -78,41 +77,51 @@ def test_client(self):
         yield self.stop_arbiter()
 
 
+TMP_FILE = './HOOK_CALLED'
+
+
 def long_hook(*args, **kw):
-    time.sleep(5)
+    with open(TMP_FILE, 'a'):
```

Comment:
I think I found a cleaner way, by creating a temporary file, and deleting it in the hook:

```patch
diff --git i/circus/tests/test_client.py w/circus/tests/test_client.py
index 0380cb3..9a56a11 100644
--- i/circus/tests/test_client.py
+++ w/circus/tests/test_client.py
@@ -1,4 +1,5 @@
 import os
+import tempfile
 
 from tornado.testing import gen_test
 from tornado.gen import coroutine, Return
@@ -77,12 +78,11 @@ class TestClient(TestCircus):
         yield self.stop_arbiter()
 
 
-TMP_FILE = './HOOK_CALLED'
+_, tmp_filename = tempfile.mkstemp(prefix='test_hook')
 
 
 def long_hook(*args, **kw):
-    with open(TMP_FILE, 'a'):
-        pass
+    os.unlink(tmp_filename)
 
 
 class TestWithHook(TestCircus):
@@ -97,13 +97,12 @@ class TestWithHook(TestCircus):
 
     @gen_test
     def test_message_id(self):
-        if os.path.exists(TMP_FILE):
-            os.unlink(TMP_FILE)
-
         hooks = {'before_stop': ('circus.tests.test_client.long_hook', False)}
         testfile, arbiter = self.run_with_hooks(hooks)
         yield arbiter.start()
         try:
+            self.assertTrue(os.path.exists(tmp_filename))
+
             msg = make_message("numwatchers")
             resp = yield self.cli.call(msg)
             self.assertEqual(resp.get("numwatchers"), 1)
@@ -118,9 +117,10 @@ class TestWithHook(TestCircus):
             resp = yield self.cli.call(make_message("numwatchers"))
             self.assertEqual(resp.get("numwatchers"), 1)
 
-            self.assertTrue(os.path.exists(TMP_FILE))
+            self.assertFalse(os.path.exists(tmp_filename))
         finally:
-            os.unlink(TMP_FILE)
+            if os.path.exists(tmp_filename):
+                os.unlink(tmp_filename)
             arbiter.stop()
```

It's not perfect but at least is doesn't create any file in the current directory :) .